cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -214,11 +217,21 @@ protected TimeSeries getTimeseries(String page, int pageSize, String names, Stri
         // Now we're going to call the retrieve_ts_out_tab function to get the data and build an
         // internal table from it so we can manipulate it further
         // This code assumes the database timezone is in UTC (per Oracle recommendation)
-        SQL retrieveSelectData = DSL.sql(
-            "table(cwms_20.cwms_ts.retrieve_ts_out_tab(?,?,cwms_20.cwms_util.to_timestamp(?),cwms_20.cwms_util.to_timestamp(?),"
-           +"'UTC',?,?,?,?,?,?,?,?) ) retrieveTs",
-            tsId,unit, beginTime.toInstant().toEpochMilli(), endTime.toInstant().toEpochMilli(),
-            null,null,null,null,null,null,null,officeId);
+        SQL retrieveSelectData = null;
+        // Query based on versionDate or query max aggregate
+        if(versionDate != null) {
```

Comment:
cwms_util.to_timestamp doesn't handle nulls so that's why you have this if/else?

Reply:
It's for the versionDate.toInstant().toEpochMilli()

Reply:
you can use a null Long class instead. Having two different conditions makes that code harder to read and maintain. Also extract variables for each of the nulls for better readability.

Reply:
Couldn't find an easy way to do this with an inline query like this. Added null handling on to_timestamp within the schema: https://bitbucket.hecdev.net/projects/CWMS/repos/cwms_database/pull-requests/324/overview